### ISSUE TITLE
[HIP-583] Allow contract deployment at a `hollow account`'s address

### ISSUE THREAD

**Issue by stoqnkpL** [hashgraph/hedera-services#4417]
`CREATE2` doesn't "play well" with `hollow` accounts. See the EET starting [here](https://github.com/hashgraph/hedera-services/blob/create2-address-clash-eet/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/Create2OperationSuite.java#L770) on branch `create2-address-clash-eet`, which:
1. Uses `CREATE2` to deploy a contract to some deterministic address `0xab...cd`
2. Deletes this contract.
3. Creates a hollow account at address `0xab...cd`
4. Repeats (1) to re-deploy a contract to `0xab...cd`

To be Ethereum-compatible, I believe we need (4) to succeed, "merging" the code of the deployed contract with the existing `hollow` account. Instead, what happens is:
- A new contract is created with id `0.0.C`; but,
- The hollow account still exists at `0.0.A` and "owns" the alias (i.e. you can do a `getAccountInfo(0xab...cd)` query and get the hollow account info; but a `getContractInfo(0xab...cd)` gets `INVALID_CONTRACT_ID`)

So in case we're trying to deploy a contract at a `hollow` account's address, we should "merge" the `hollow` account into a contract, set its `code`, and a default `JContractIDKey`, completing the `hollow` account creation with this step. This should be safe since the `hollow` account hasn't been the `payer` of any transactions, otherwise, it wouldn't be `hollow`.

But if the account isn't `hollow`, we should not allow the "merging" even if its nonce is `0`, since this account could have signed and executed all kinds of `HAPI` transactions and there could be side effects of converting it to a contract.

So basically that would mean that a `hollow` account in Hedera could end up being a `regular account` OR a `contract account`.
